front/lib/api/assistant/configuration.ts

```diff
@@ -1504,6 +1509,24 @@ async function _createAgentDataSourcesConfigData(
   const results = await Promise.all(getDataSourcesQueries);
   const dataSources = results.flat();
 
+  // Since the UI does not currently provide the data source view,
+  // we retrieve the view associated with the data from the global vault
+  // and assign it to the agent data source configuration.
+  const uniqueDataSources = _.uniqBy(dataSources, (ds) => ds.id);
+  const globalVault = await VaultResource.fetchWorkspaceGlobalVault(auth);
+  const dataSourceViews = (
+    await Promise.all(
+      uniqueDataSources.map((ds) => {
+        return DataSourceViewResource.listForDataSourceInVault(
+          auth,
+          renderDataSourceType(ds),
+          globalVault,
+          { transaction: t }
+        );
+      })
```

Comment:
Can't we do a single query by passing a list of datasources (datasourceIds) ? `listForDataSourcesInVault`

Reply:
Done.